examples/conditional-request-handling/JsonRequestHandler.cpp

```diff
@@ -2,29 +2,27 @@
 
 #include "JsonRequestHandler.h"
 #include <string>
-#include "ConditionalRequestSubscription.h"
+#include "yarpl/v/Flowable.h"
+#include "yarpl/v/Flowables.h"
 
-using namespace ::reactivesocket;
+using namespace reactivesocket;
+using namespace rsocket;
+using namespace yarpl;
 
 /// Handles a new inbound Stream requested by the other end.
-void JsonRequestHandler::handleRequestStream(
-    Payload request,
-    StreamId streamId,
-    const std::shared_ptr<Subscriber<Payload>>& response) noexcept {
+yarpl::Reference<yarpl::Flowable<reactivesocket::Payload>>
+JsonRequestHandler::handleRequestStream(Payload request, StreamId streamId) {
   LOG(INFO) << "JsonRequestHandler.handleRequestStream " << request;
 
   // string from payload data
-  auto pds = request.moveDataToString();
-  auto requestString = std::string(pds, request.data->length());
+  const char* p = reinterpret_cast<const char*>(request.data->data());
```

Comment:
I don't understand the question in light of this portion of code. 

I created cloneDataToString() because I couldn't get it to work in places where const was required, so I needed a method that copied instead of moved from a const Payload. 

The code in question is in ExampleSubscriber:

```cpp
void ExampleSubscriber::onNext(const Payload& element) noexcept {
```

That expects a const ref, and because of that I can't move data from it.

![screen shot 2017-04-25 at 1 16 02 pm](https://cloud.githubusercontent.com/assets/813492/25405815/5e1be3b8-29b9-11e7-9105-dc6c3afa2467.png)